src/btf.cpp

```diff
@@ -450,30 +451,34 @@ Struct BTF::resolve_args(const std::string &func, bool ret)
 
   Struct args(0, false);
   int j = 0;
+  int arg_idx = 0;
+  __u32 type_size = 0;
   for (; j < vlen; j++, p++)
   {
     const char *str = btf_str(func_id.btf, p->name_off);
     if (!str)
       throw std::runtime_error("failed to resolve arguments");
 
     SizedType stype = get_stype(BTFId{ .btf = func_id.btf, .id = p->type });
-    stype.funcarg_idx = j;
+    stype.funcarg_idx = arg_idx;
     stype.is_funcarg = true;
     args.AddField(str, stype, args.size, std::nullopt, false);
     // kfunc (fentry/fexit) args are stored in a u64 array.
     // Note that it's ok to represent them by a struct as we will use GEP with
     // funcarg_idx to access them in codegen.
-    args.size += 8;
+    type_size = btf__resolve_size(func_id.btf, p->type);
+    args.size += type_size;
+    arg_idx += std::ceil((float)type_size / (float)8);
```

Comment:
Is using the float to handle `type_size < 8`? If so, maybe more clear to assert that `type_size == 8 || type_size == 16`. Would be clear if kernel ever accidentally changes its ABI that a breakage occurred.

Reply:
We already know that the type_size can be 4 (this is expected behavior)

Reply:
Oh right, slot size is 8 or 16, not type